File: src/models/Alternativa.py
from src.connections.DatabaseConnection import db


class Alternativa(db.Model):
    __tablename__ = "alternativas"

    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    texto = db.Column(db.Text, nullable=False)
    correta = db.Column(db.Boolean, nullable=False)
    justificativa = db.Column(db.Text, nullable=False)
    questao_id = db.Column(db.Integer, db.ForeignKey("questoes.id"), nullable=False)

    questao = db.relationship("Questao", back_populates="alternativas")

File: templates/src/models/Alternativa.ts
import type AlternativaI from '@/interfaces/Alternativa'

export default class Alternativa {
  id: number
  texto: string
  correta: boolean
  justificativa: string
  selecionada: boolean

  constructor(alternativa: AlternativaI) {
    this.id = alternativa.id ? alternativa.id : 0
    this.texto = alternativa.texto
    this.correta = alternativa.correta
    this.justificativa = alternativa.justificativa
    this.selecionada = false
  }

  selecionar(): void {
    this.selecionada = true
  }
}
